### PR TITLE
no-jira: Collect bootstrap logs when control plane provisioning fails

### PR DESCRIPTION
**Pull request by patrickdillon**
This updates the installer to gather and analyze logs from the bootstrap node (and serial nodes) when provisioning of control plane machines fail. Investigating [OCPBUGS-43625](https://issues.redhat.com/browse/OCPBUGS-43625), it would be helpful to have the logs from the bootstrap machine. So let's do this automatically

### CODE REVIEW

**Comment by patrickdillon** [comment]
Test results look good. e2e-aws-ovn collected three log bundles. e2e-azure-ovn collected one, because runs 2 & 3 fail before machine provisioning.

**Comment by patrickdillon** [comment]
Dropped DNM commit. ready for review

**Comment by patrickdillon** [comment]
Want to see about how we can get support for baremetal ipi here

File: pkg/utils/baremetal/bootstrap.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ import (
 	"path/filepath"
 
 	baremetalhost "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	"github.com/pkg/errors"
```
Comment (r4f4) [inline pkg/utils/baremetal/bootstrap.go:11]:
```suggestion

```
Let's not introduce this deprecated package in new code

File: pkg/utils/baremetal/bootstrap.go [hunk 2]
```diff
@@ -106,5 +108,5 @@ func WaitForBaremetalBootstrapControlPlane(ctx context.Context, config *rest.Con
 		return fmt.Errorf("failed to persist masters file to disk: %w", err)
 	}
 
-	return withSyncErr
+	return errors.Wrap(withSyncErr, asset.ControlPlaneCreationError)
```
Comment (r4f4) [inline pkg/utils/baremetal/bootstrap.go:111]:
```suggestion
	return fmt.Errorf("%s: %w", asset.ControlPlaneCreationError, withSyncErr)
```

Comment (patrickdillon) [reply pkg/utils/baremetal/bootstrap.go:111]: d'oh how embarrassing, as I know better. The problem is we need to return `nil` if `withSyncErr` is `nil`, so I think we need to go a bit further than the suggestion?

Comment (r4f4) [reply pkg/utils/baremetal/bootstrap.go:111]: Ah yes.
```suggestion
    if withSyncErr != nil {
	    return fmt.Errorf("%s: %w", asset.ControlPlaneCreationError, withSyncErr)
	}
	return nil
```

Comment (patrickdillon) [reply pkg/utils/baremetal/bootstrap.go]: should be fixed now

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Pull request merged**